antarest/study/dao/study_conversion/study_converter.py

```diff
@@ -0,0 +1,329 @@
+            self._new_dao.save_thermal_prepro(area_id, thermal_id, prepro_id)
+
+            modulation_id = self._matrix_service.create(self._source_dao.get_thermal_modulation(area_id, thermal_id))
+            self._new_dao.save_thermal_modulation(area_id, thermal_id, modulation_id)
+
+            series_id = self._matrix_service.create(self._source_dao.get_thermal_series(area_id, thermal_id))
+            self._new_dao.save_thermal_series(area_id, thermal_id, series_id)
+
+            if self._study_version >= STUDY_VERSION_8_7:
+                fuel_cost_id = self._matrix_service.create(self._source_dao.get_thermal_fuel_cost(area_id, thermal_id))
+                self._new_dao.save_thermal_fuel_cost(area_id, thermal_id, fuel_cost_id)
+
+                co2_cost_id = self._matrix_service.create(self._source_dao.get_thermal_co2_cost(area_id, thermal_id))
+                self._new_dao.save_thermal_co2_cost(area_id, thermal_id, co2_cost_id)
+
+    def _convert_renewable_clusters(self, area_id: str, renewables: Sequence[RenewableCluster]) -> None:
+        self._new_dao.save_renewables(area_id, renewables)
+        for renewable in renewables:
+            renewable_id = renewable.id.lower()
+            series_id = self._matrix_service.create(self._source_dao.get_renewable_series(area_id, renewable_id))
+            self._new_dao.save_renewable_series(area_id, renewable_id, series_id)
+
+    def _convert_short_term_storages(
+        self, area_id: str, storages: list[STStorage], constraints: dict[str, list[STStorageAdditionalConstraint]]
+    ) -> None:
+        self._new_dao.save_st_storages(area_id, storages)
+        for st_storage in storages:
+            sts_id = st_storage.id
+            p_max_inj_id = self._matrix_service.create(self._source_dao.get_st_storage_pmax_injection(area_id, sts_id))
+            self._new_dao.save_st_storage_pmax_injection(area_id, sts_id, p_max_inj_id)
+
+            p_max_wit_id = self._matrix_service.create(self._source_dao.get_st_storage_pmax_withdrawal(area_id, sts_id))
+            self._new_dao.save_st_storage_pmax_withdrawal(area_id, sts_id, p_max_wit_id)
+
+            low_id = self._matrix_service.create(self._source_dao.get_st_storage_lower_rule_curve(area_id, sts_id))
+            self._new_dao.save_st_storage_lower_rule_curve(area_id, sts_id, low_id)
+
+            upper_id = self._matrix_service.create(self._source_dao.get_st_storage_upper_rule_curve(area_id, sts_id))
+            self._new_dao.save_st_storage_upper_rule_curve(area_id, sts_id, upper_id)
+
+            inflows_id = self._matrix_service.create(self._source_dao.get_st_storage_inflows(area_id, sts_id))
+            self._new_dao.save_st_storage_inflows(area_id, sts_id, inflows_id)
+
+            if self._study_version >= STUDY_VERSION_9_2:
+                self._new_dao.save_st_storage_additional_constraints(area_id, sts_id, constraints[sts_id])
```

Comment:
`constraints[sts_id]` can raise a `KeyError`. You can handle it with `constraints.get(sts_id, []`